ui/src/data_explorer/components/QueryEditor.js

```diff
@@ -69,55 +198,102 @@ const QueryEditor = React.createClass({
           onChange={this.handleChange}
           onKeyDown={this.handleKeyDown}
           onBlur={this.handleUpdate}
-          ref={editor => (this.editor = editor)}
+          ref={editor => this.editor = editor}
           value={value}
           placeholder="Enter a query or select database, measurement, and field below and have us build one for you..."
           autoComplete="off"
           spellCheck="false"
         />
-        {this.renderStatus(status)}
-        <Dropdown
-          items={QUERY_TEMPLATES}
-          selected={'Query Templates'}
-          onChoose={this.handleChooseTemplate}
-          className="query-editor--templates"
-        />
+        <div
+          className={classNames('varmoji', {'varmoji-rotated': isTemplating})}
+        >
+          <div className="varmoji-container">
```

Comment:
I wonder if it's incorrect to attach the `varmoji` term this far up the tree -- `varmoji-container` contains both varmoji behavior as well as indicating query status, which is a separate concept.

It looks like we already use the term `drawer` to refer to this area, i.e. opening/closing the drawer, controls the value of `isTemplating`. Maybe use `drawer` or a variant thereof for the namespace here?